nemo_aligner/models/mm/stable_diffusion/megatron_sd_draftp_model.py

```diff
@@ -0,0 +1,468 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Mapping
+
+import numpy as np
+import torch
+import wandb
+from omegaconf.dictconfig import DictConfig
+from PIL import Image
+from tqdm import tqdm
+
+import nemo.collections.multimodal.parts.stable_diffusion.pipeline as sampling_utils
+from nemo.collections.multimodal.models.text_to_image.stable_diffusion.ldm.ddpm import LatentDiffusion
+from nemo.collections.nlp.modules.common.megatron.utils import average_losses_across_data_parallel_group
+from nemo.collections.nlp.parts.utils_funcs import get_last_rank
+from nemo_aligner.models.alignable_interface import AlignableGenerativeInterface
+from nemo_aligner.utils.train_utils import prepare_for_training_step
+from nemo_aligner.utils.utils import configure_batch_sizes
+
+try:
+    from megatron.core import parallel_state
+    from megatron.core.pipeline_parallel.schedules import get_forward_backward_func
+
+    HAVE_MEGATRON_CORE = True
+except (ImportError, ModuleNotFoundError):
+    HAVE_MEGATRON_CORE = False
+
+BatchType = Mapping[str, torch.tensor]
+
+
+class MegatronSDDRaFTPModel(AlignableGenerativeInterface):
```

Comment:
this should inherit from a MegatronGPTModel like object, just like [MegatronDPOModel](https://github.com/NVIDIA/NeMo-Aligner/blob/main/nemo_aligner/models/nlp/gpt/megatron_gpt_dpo_model.py#L44)

Reply:
I think lots of methods in the GPT model are redundant for the SD model, plus intuitively, it doesn't make sense if an SD model inherits from a GPT class

Reply:
fully changed interface